File: lldb/test/lang/objc/real-definition/Foo.m
#import "Foo.h"

@implementation Foo

- (id)init
{
    self = [super init];
    if (self) {
        _bar = [[Bar alloc] init];
    }
    return self; // Set breakpoint where Bar is an interface
}

- (void)dealloc
{
    [_bar release];
    [super dealloc];
}

- (NSString *)description
{
    return [NSString stringWithFormat:@"%p: @Foo { _bar = %@ }", self, _bar];
}

@end

File: lldb/lit/Reproducer/Inputs/foo.cpp
struct Foo {
  Foo();
  Foo(int i);

  void A();
  void B(int i);
  int C(int i);
  int D(bool b) const;
  static void E();
  static int F(int i);
};

void Foo::A() {}
void Foo::B(int i) {}
int Foo::C(int i) { return i; }
int Foo::D(bool b) const { return 1; }
void Foo::E() {}
int Foo::F(int i) { return i; }
